### PR TITLE
add function to retreive index's section memory size and use this to get symbol table size

### PR DESCRIPTION
**Pull request by sipian**
[#244](https://github.com/prometheus/tsdb/issues/244)
I have added a function in index/index.go to get a section's size. 
Acc to documentation of [index.md](https://github.com/prometheus/tsdb/blob/master/docs/format/index.md) & [this](https://github.com/prometheus/tsdb/blob/master/index/index.go#L693) comment, a sections 1st 4 bytes holds the big endian encoded content length .

For symbol table size I retrieve these 4 bytes from (r.toc.symbols offsets)

### CODE REVIEW

File: index/index.go [hunk 1]
```diff
@@ -743,6 +743,27 @@ func (r *Reader) decbufUvarintAt(off int) decbuf {
 	return dec
 }
 
+// getSectionSize returns the first 4 bytes in a section starting at offset off
+// to get the content length bytes
+func (r *Reader) getSectionSize(off int) int {
+	if off == 0 {
```
Comment (Gouthamve) [inline index/index.go]: You can remove this check.

File: index/index.go [hunk 2]
```diff
@@ -743,6 +743,27 @@ func (r *Reader) decbufUvarintAt(off int) decbuf {
...
+		return 0
+	}
+	b := r.b.Range(off, off+4)
+	var l int
+	if l <= 0 {
```
Comment (Gouthamve) [inline index/index.go]: Umm, `l` has just been declared hence will always be 0.

File: index/index.go [hunk 3]
```diff
@@ -743,6 +743,27 @@ func (r *Reader) decbufUvarintAt(off int) decbuf {
...
+		l = -1
+	} else {
+		l  = int(binary.BigEndian.Uint32(b))
+	}
+	return l
+}
+
+// getSymbolTableSize returns the bytes taken by the symbol table of Reader object
```
Comment (Gouthamve) [inline index/index.go]: end comments with `.`

**Comment by sipian** [comment]
@fabxc @brian-brazil 

I have facing some difficulty in exposing symbol-table len to IndexReader interface.
The way I am doing is in **index.go** I've added a function GetSymbolTableSize() in struct Reader.
I need to call this function from each Block in db.block. Hence, I get IndexReader for each block(using Index()) and have added GetSymbolTableSize() in IndexReader interface in **block.go**.
But then we need to add implementation of GetSymbolTableSize() in head.go & querier_test.go() also.
Could you suggest a way to expose  GetSymbolTableSize() to IndexReader interface?

File: block.go [hunk 4]
```diff
@@ -84,6 +84,9 @@ type IndexReader interface {
 	// LabelIndices returns a list of string tuples for which a label value index exists.
 	LabelIndices() ([][]string, error)
 
+	// GetSymbolTableSize returns the size occupied by the symbol table of Reader object.
+	GetSymbolTableSize() uint32
+
```
Comment (fabxc) [inline block.go]: This has nothing to do with functionality of an `IndexReader` and makes strong assumptions about how it is implemented. However we want to expose it, this interface isn't it.

Comment (sipian) [reply block.go]: I had the same question in this https://github.com/prometheus/tsdb/pull/297 PR.
There I have added a new interface `IndexExtra` for exposing LabelName() function.
Does this way work or is there any other way to expose functions from index.go ?

**Comment by codesome** [comment]
@sipian, do you plan to work on this?

**Comment by sipian** [comment]
@codesome 
You can go ahead

**Comment by krasi-georgiev** [comment]
Superseded by https://github.com/prometheus/tsdb/pull/375

**Pull request closed without merging**